dash_table/Format.py

```diff
@@ -0,0 +1,286 @@
+        if value not in Scheme:
+            raise Exception('expected value to be one of', str(list(Scheme)))
+
+        self._specifier['type'] = value
+        return self
+
+    def sign(self, value):
+        if value not in Sign:
+            raise Exception('expected value to be one of', str(list(Sign)))
+
+        self._specifier['sign'] = value
+        return self
+
+    def symbol(self, value):
+        if value not in Symbol:
+            raise Exception('expected value to be one of', str(list(Symbol)))
+
+        self._specifier['symbol'] = value
+        return self
+
+    def trim(self, value):
+        if isinstance(value, bool):
+            value = Trim.yes if value else Trim.no
+
+        if value not in Trim:
+            raise Exception('expected value to be one of', str(list(Trim)))
+
+        self._specifier['trim'] = value
+        return self
+
+    def symbol_prefix(self, value):
+        if not is_string(value):
+            raise Exception('expected value to be a string')
+
+        if 'symbol' not in self._locale:
+            self._locale['symbol'] = [value, '']
+        else:
+            self._locale['symbol'][0] = value
+
+        return self
+
+    # Locale
+    def symbol_suffix(self, value):
+        if not is_string(value):
+            raise Exception('expected value to be a string')
+
+        if 'symbol' not in self._locale:
+            self._locale['symbol'] = ['', value]
+        else:
+            self._locale['symbol'][1] = value
+
+        return self
+
+    def decimal_delimitor(self, value):
+        if not is_string(value):
+            raise Exception('expected value to be a string')
+
+        self._locale['decimal'] = value
+        return self
+
+    def group_delimitor(self, value):
+        if not is_string(value):
+            raise Exception('expected value to be a string')
+
+        self._locale['group'] = value
+        return self
+
+    def groups(self, groups):
+        groups = groups if isinstance(groups, list) else [groups] if isinstance(groups, int) else None
+
+        if not isinstance(groups, list) or len(groups) == 0:
+            raise Exception('expected groups to be an integer or a list of integers')
+
+        for group in groups:
+            if not isinstance(group, int) or group <= 0:
+                raise Exception('expected entry to be a non-negative integer')
+
+        self._locale['grouping'] = groups
+        return self
+
+    # Nully
+    def nully(self, value):
+        self._nully = value
+        return self
+
+    # Prefix
+    def si_prefix(self, value):
+        if value not in Prefix:
+            raise Exception('expected value to be one of', str(list(Prefix)))
+
+        self._prefix = value
+        return self
+
+    def to_plotly_json(self):
+        f = {}
+        f['locale'] = self._locale.copy()
+        f['nully'] = self._nully
+        f['prefix'] = self._prefix
+        f['specifier'] = '{0}{1}{2}{3}{4}{5}{6}{7}{8}{9}'.format(
```

Comment:
minor - you can omit the numbers here... `'{}{}{}{}{}{}{}{}{}{}'`

It's annoying that `fill` has extra logic, without that you could do
`'{fill}{align}{sign}...'.format(**self._specifier)`

I suppose it would work to do something like:
```py
'{fill_if}{align}{sign}...'.format(
    fill_if=self._specifier['fill'] if self._specifier['align'] != Align.default else '',
    **self._specifier
)
```